internal/image/oci.go

```diff
@@ -143,13 +143,26 @@ func (m *manifestOCI1) UpdatedImageNeedsLayerDiffIDs(options types.ManifestUpdat
 // if the combination of CompressionOperation and CompressionAlgorithm specified
 // in one or more options.LayerInfos items indicates that a layer is compressed using
 // an algorithm that is not allowed in OCI.
-func (m *manifestOCI1) UpdatedImage(ctx context.Context, options types.ManifestUpdateOptions) (types.Image, error) {
+func (m *manifestOCI1) UpdatedImage(ctx context.Context, options types.ManifestUpdateOptions) (image types.Image, retErr error) {
 	copy := manifestOCI1{ // NOTE: This is not a deep copy, it still shares slices etc.
 		src:        m.src,
 		configBlob: m.configBlob,
 		m:          manifest.OCI1Clone(m.m),
 	}
 
+	manifestTmp := m
+	defer func() {
+		if retErr != nil {
+			m = manifestTmp
+		}
+	}()
```

Comment:
I don’t understand at all what this is intended to do. `m` is local to this function, and just a pointer. Assigning to it, AFAICS, does nothing visible to the caller.